// material-ui-player/dist/stories/public/VideoCard.stories.js
"use strict";
var __assign = (this && this.__assign) || function () {
    __assign = Object.assign || function(t) {
        for (var s, i = 1, n = arguments.length; i < n; i++) {
            s = arguments[i];
            for (var p in s) if (Object.prototype.hasOwnProperty.call(s, p))
                t[p] = s[p];
        }
        return t;
    };
    return __assign.apply(this, arguments);
};
var __importDefault = (this && this.__importDefault) || function (mod) {
    return (mod && mod.__esModule) ? mod : { "default": mod };
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.FirebaseStorage = exports.Default = void 0;
var react_1 = __importDefault(require("react"));
var VideoCard_1 = require("../../components/VideoCard");
exports.default = {
    title: 'Components/VideoCard',
    component: VideoCard_1.VideoCard,
    argTypes: {
        color: {
            defaultValue: 'primary',
            control: {
                type: 'radio',
            },
            options: ['primary', 'secondary']
        },
        thickness: {
            defaultValue: 'medium',
            control: {
                type: 'radio',
            },
            options: ['thin', 'medium', 'large']
        },
        onForwardClick: {
            action: 'forward'
        },
        onBackwardClick: {
            action: 'backward'
        },
        onEnded: {
            action: 'end'
        },
        forward: {
            defaultValue: true,
            control: {
                type: 'radio',
            },
            options: [true, false]
        },
        backward: {
            defaultValue: true,
            control: {
                type: 'radio',
            },
            options: [true, false]
        },
        mute: {
            defaultValue: true,
            control: {
                type: 'radio',
            },
            options: [true, false]
        },
        speed: {
            defaultValue: true,
            control: {
                type: 'radio',
            },
            options: [true, false]
        },
        loop: {
            defaultValue: false,
            control: {
                type: 'radio',
            },
            options: [true, false]
        },
    }
};
var Template = function (args) { return react_1.default.createElement(VideoCard_1.VideoCard, __assign({}, args)); };
exports.Default = Template.bind({});
exports.Default.args = {
    src: 'https://sample-videos.com/video123/mp4/720/big_buck_bunny_720p_1mb.mp4',
    width: 500,
};
exports.FirebaseStorage = Template.bind({});
exports.FirebaseStorage.args = {
    src: new Promise(function (resolve) {
        //simulate get downloadURL
        setTimeout(function () {
            resolve('https://sample-videos.com/video123/mp4/720/big_buck_bunny_720p_1mb.mp4');
        }, 500);
    }),
    width: 500,
};
