// material-ui-player/dist/stories/ControlKeys.stories.js
"use strict";
var __assign = (this && this.__assign) || function () {
    __assign = Object.assign || function(t) {
        for (var s, i = 1, n = arguments.length; i < n; i++) {
            s = arguments[i];
            for (var p in s) if (Object.prototype.hasOwnProperty.call(s, p))
                t[p] = s[p];
        }
        return t;
    };
    return __assign.apply(this, arguments);
};
var __importDefault = (this && this.__importDefault) || function (mod) {
    return (mod && mod.__esModule) ? mod : { "default": mod };
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.NoBackwardAndForward = exports.Playing = exports.Default = void 0;
var react_1 = __importDefault(require("react"));
var components_1 = require("../components");
exports.default = {
    title: 'Components/ControlKeys',
    component: components_1.ControlKeys,
    argTypes: {
        onPauseClick: {
            action: 'pause'
        },
        onPlayClick: {
            action: 'play'
        },
        onForwardClick: {
            action: 'forward'
        },
        onBackwardClick: {
            action: 'backward'
        },
        onStopClick: {
            action: 'stop'
        },
        color: {
            defaultValue: 'primary',
            control: {
                type: 'radio',
            },
            options: ['primary', 'secondary']
        },
    }
};
var Template = function (args) { return react_1.default.createElement(components_1.ControlKeys, __assign({}, args)); };
exports.Default = Template.bind({});
exports.Default.args = {
    backward: true,
    forward: true,
    playing: false
};
exports.Playing = Template.bind({});
exports.Playing.args = {
    backward: true,
    forward: true,
    playing: true
};
exports.NoBackwardAndForward = Template.bind({});
exports.NoBackwardAndForward.args = {
    playing: false
};
