// material-ui-player/dist/stories/VolumeBar.stories.js
"use strict";
var __assign = (this && this.__assign) || function () {
    __assign = Object.assign || function(t) {
        for (var s, i = 1, n = arguments.length; i < n; i++) {
            s = arguments[i];
            for (var p in s) if (Object.prototype.hasOwnProperty.call(s, p))
                t[p] = s[p];
        }
        return t;
    };
    return __assign.apply(this, arguments);
};
var __importDefault = (this && this.__importDefault) || function (mod) {
    return (mod && mod.__esModule) ? mod : { "default": mod };
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.HideMute = exports.Default = void 0;
var react_1 = __importDefault(require("react"));
var components_1 = require("../components");
exports.default = {
    title: 'Components/VolumeBar',
    component: components_1.VolumeBar,
    argTypes: {
        color: {
            defaultValue: 'primary',
            control: {
                type: 'radio',
            },
            options: ['primary', 'secondary']
        },
        mute: {
            defaultValue: true,
            control: {
                type: 'radio',
            },
            options: [true, false]
        },
        thickness: {
            defaultValue: 'medium',
            control: {
                type: 'radio',
            },
            options: ['thin', 'medium', 'large']
        }
    }
};
var mediaPlayer = new Audio();
var Template = function (args) { return react_1.default.createElement(components_1.VolumeBar, __assign({}, args)); };
exports.Default = Template.bind({});
exports.Default.args = {
    player: mediaPlayer,
};
exports.HideMute = Template.bind({});
exports.HideMute.args = {
    player: mediaPlayer,
    mute: false,
};
