// material-ui-player/dist/components/MaterialUIAudio.js
"use strict";
var __createBinding = (this && this.__createBinding) || (Object.create ? (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    Object.defineProperty(o, k2, { enumerable: true, get: function() { return m[k]; } });
}) : (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    o[k2] = m[k];
}));
var __setModuleDefault = (this && this.__setModuleDefault) || (Object.create ? (function(o, v) {
    Object.defineProperty(o, "default", { enumerable: true, value: v });
}) : function(o, v) {
    o["default"] = v;
});
var __importStar = (this && this.__importStar) || function (mod) {
    if (mod && mod.__esModule) return mod;
    var result = {};
    if (mod != null) for (var k in mod) if (k !== "default" && Object.prototype.hasOwnProperty.call(mod, k)) __createBinding(result, mod, k);
    __setModuleDefault(result, mod);
    return result;
};
var __awaiter = (this && this.__awaiter) || function (thisArg, _arguments, P, generator) {
    function adopt(value) { return value instanceof P ? value : new P(function (resolve) { resolve(value); }); }
    return new (P || (P = Promise))(function (resolve, reject) {
        function fulfilled(value) { try { step(generator.next(value)); } catch (e) { reject(e); } }
        function rejected(value) { try { step(generator["throw"](value)); } catch (e) { reject(e); } }
        function step(result) { result.done ? resolve(result.value) : adopt(result.value).then(fulfilled, rejected); }
        step((generator = generator.apply(thisArg, _arguments || [])).next());
    });
};
var __generator = (this && this.__generator) || function (thisArg, body) {
    var _ = { label: 0, sent: function() { if (t[0] & 1) throw t[1]; return t[1]; }, trys: [], ops: [] }, f, y, t, g;
    return g = { next: verb(0), "throw": verb(1), "return": verb(2) }, typeof Symbol === "function" && (g[Symbol.iterator] = function() { return this; }), g;
    function verb(n) { return function (v) { return step([n, v]); }; }
    function step(op) {
        if (f) throw new TypeError("Generator is already executing.");
        while (_) try {
            if (f = 1, y && (t = op[0] & 2 ? y["return"] : op[0] ? y["throw"] || ((t = y["return"]) && t.call(y), 0) : y.next) && !(t = t.call(y, op[1])).done) return t;
            if (y = 0, t) op = [op[0] & 2, t.value];
            switch (op[0]) {
                case 0: case 1: t = op; break;
                case 4: _.label++; return { value: op[1], done: false };
                case 5: _.label++; y = op[1]; op = [0]; continue;
                case 7: op = _.ops.pop(); _.trys.pop(); continue;
                default:
                    if (!(t = _.trys, t = t.length > 0 && t[t.length - 1]) && (op[0] === 6 || op[0] === 2)) { _ = 0; continue; }
                    if (op[0] === 3 && (!t || (op[1] > t[0] && op[1] < t[3]))) { _.label = op[1]; break; }
                    if (op[0] === 6 && _.label < t[1]) { _.label = t[1]; t = op; break; }
                    if (t && _.label < t[2]) { _.label = t[2]; _.ops.push(op); break; }
                    if (t[2]) _.ops.pop();
                    _.trys.pop(); continue;
            }
            op = body.call(thisArg, _);
        } catch (e) { op = [6, e]; y = 0; } finally { f = t = 0; }
        if (op[0] & 5) throw op[1]; return { value: op[0] ? op[1] : void 0, done: true };
    }
};
Object.defineProperty(exports, "__esModule", { value: true });
var react_1 = __importStar(require("react"));
var core_1 = require("@material-ui/core");
var utils_1 = require("../lib/utils");
var index_1 = require("./index");
var types_1 = require("../types");
var styles_1 = require("@material-ui/core/styles");
var useStyles = styles_1.makeStyles(function (theme) {
    var _a;
    return styles_1.createStyles({
        container: {
            display: 'flex',
            flexDirection: 'column',
            alignItems: 'center',
            paddingTop: theme.spacing(1),
            paddingBottom: theme.spacing(1),
            '&:last-child': {
                paddingBottom: theme.spacing(1),
            },
        },
        card: {
            width: function (props) { return props.width || '100%'; },
        },
        audio: {
            display: 'none',
        },
        controls: (_a = {
                display: 'flex',
                alignItems: 'center'
            },
            _a[theme.breakpoints.down('sm')] = {
                justifyContent: 'space-around',
            },
            _a),
    });
});
var MaterialUIAudio = function (props) {
    var _a = react_1.useState(''), url = _a[0], setUrl = _a[1];
    var _b = react_1.useState(false), playing = _b[0], setPlaying = _b[1];
    var _c = react_1.useState({}), time = _c[0], setTime = _c[1];
    var _d = react_1.useState(false), muted = _d[0], setMuted = _d[1];
    var _e = react_1.useState(0), progress = _e[0], setProgress = _e[1];
    var _f = react_1.useState(null), playerTimeout = _f[0], setPlayerTimeout = _f[1];
    var refPlayer = react_1.useRef(null);
    var player = new types_1.Player(refPlayer);
    var pausePlaying = react_1.useCallback(function () {
        player.pause();
        playerTimeout && clearInterval(playerTimeout);
        setPlaying(false);
    }, [playerTimeout]);
    var setTimeProgress = react_1.useCallback(function (value) {
        var currentTime = (value / 100) * player.duration;
        player.currentTime = currentTime;
        var progressTime = {
            currentTime: player.currentTime,
            duration: player.duration,
        };
        setTime(progressTime);
        setProgress((progressTime.currentTime / progressTime.duration) * 100);
    }, []);
    var stop = react_1.useCallback(function () {
        pausePlaying();
        setTimeProgress(0);
    }, [pausePlaying, setTimeProgress]);
    var src = props.src, _g = props.onEnded, onEnded = _g === void 0 ? stop : _g, _h = props.onBackwardClick, onBackwardClick = _h === void 0 ? function () { } : _h, _j = props.onForwardClick, onForwardClick = _j === void 0 ? function () { } : _j, width = props.width;
    var classes = useStyles({ width: width });
    var intervalCheck = react_1.useCallback(function () {
        if (player.ended) {
            onEnded();
            playerTimeout && clearInterval(playerTimeout);
            return;
        }
        var progressTime = {
            currentTime: player.currentTime,
            duration: player.duration,
        };
        setTime(progressTime);
        setProgress((progressTime.currentTime / progressTime.duration) * 100);
    }, [onEnded]);
    var onPlay = react_1.useCallback(function () { return __awaiter(void 0, void 0, void 0, function () {
        var audioUrl, _a;
        return __generator(this, function (_b) {
            switch (_b.label) {
                case 0:
                    if (!!url) return [3 /*break*/, 4];
                    if (!(typeof src === 'string')) return [3 /*break*/, 1];
                    _a = src;
                    return [3 /*break*/, 3];
                case 1: return [4 /*yield*/, src];
                case 2:
                    _a = _b.sent();
                    _b.label = 3;
                case 3:
                    audioUrl = _a;
                    setUrl(audioUrl);
                    player.src = audioUrl;
                    player.load();
                    _b.label = 4;
                case 4:
                    setPlaying(true);
                    setPlayerTimeout(setInterval(intervalCheck, 50));
                    player.play();
                    return [2 /*return*/];
            }
        });
    }); }, [intervalCheck, url, src]);
    react_1.useEffect(function () {
        player.autoplay = !!props.autoplay;
        player.loop = !!props.loop;
        stop();
    }, [src, props.autoplay, props.loop]);
    var onVolumeClick = react_1.useCallback(function (volume) {
        player.volume = volume;
    }, []);
    var onMute = react_1.useCallback(function () {
        player.muted = !muted;
        setMuted(!muted);
    }, [muted]);
    var onProgressClick = react_1.useCallback(function (value) { return __awaiter(void 0, void 0, void 0, function () {
        var audioUrl, _a;
        return __generator(this, function (_b) {
            switch (_b.label) {
                case 0:
                    if (!!player.src) return [3 /*break*/, 4];
                    if (!(typeof src === 'string')) return [3 /*break*/, 1];
                    _a = src;
                    return [3 /*break*/, 3];
                case 1: return [4 /*yield*/, src];
                case 2:
                    _a = _b.sent();
                    _b.label = 3;
                case 3:
                    audioUrl = _a;
                    setUrl(audioUrl);
                    player.src = audioUrl;
                    player.load();
                    player.onMetadata = function () { return setTimeProgress(value); };
                    return [2 /*return*/];
                case 4:
                    setTimeProgress(value);
                    return [2 /*return*/];
            }
        });
    }); }, [src, setTimeProgress]);
    var onSpeedChange = react_1.useCallback(function (speed) {
        player.speed = speed;
    }, []);
    return (react_1.default.createElement(core_1.Card, { className: classes.card },
        react_1.default.createElement(core_1.CardContent, { className: classes.container },
            react_1.default.createElement(core_1.Grid, { container: true, justify: "center", alignItems: "center" },
                react_1.default.createElement("audio", { ref: refPlayer, className: classes.audio }, url && react_1.default.createElement("source", { type: utils_1.getMimeType(url), src: url })),
                react_1.default.createElement(core_1.Grid, { item: true, xs: 12 },
                    react_1.default.createElement(index_1.Progress, { progress: progress, onProgressClick: function (v) { return __awaiter(void 0, void 0, void 0, function () { return __generator(this, function (_a) {
                            switch (_a.label) {
                                case 0: return [4 /*yield*/, onProgressClick(v)];
                                case 1: return [2 /*return*/, _a.sent()];
                            }
                        }); }); } })),
                react_1.default.createElement(core_1.Grid, { item: true, xs: 12, sm: 6 },
                    react_1.default.createElement("div", { className: classes.controls },
                        react_1.default.createElement(index_1.MediaTime, { time: time }),
                        react_1.default.createElement(index_1.ControlKeys, { onPauseClick: pausePlaying, onPlayClick: onPlay, backward: props.backward, forward: props.forward, playing: playing, onForwardClick: onForwardClick, onBackwardClick: onBackwardClick, onStopClick: stop }))),
                props.speed &&
                    react_1.default.createElement(core_1.Grid, { item: true, xs: 6, sm: 3 },
                        react_1.default.createElement(index_1.SpeedBar, { onSpeedChange: onSpeedChange })),
                react_1.default.createElement(core_1.Grid, { item: true, xs: props.speed ? 6 : 12, sm: props.speed ? 3 : 6 },
                    react_1.default.createElement(index_1.VolumeBar, { onMuteClick: onMute, onVolumeClick: onVolumeClick }))))));
};
exports.default = MaterialUIAudio;
